usr/sbin/rear

```diff
@@ -58,8 +58,13 @@ readonly PROGRAM=${0##*/}
 readonly VERSION=2.6
 readonly RELEASE_DATE=Git
 
-# Used in framework-functions.sh to calculate the time spent executing rear:
-readonly STARTTIME=$SECONDS
+# Set machine readable and human readable start date and time values.
+# START_SECONDS is set to have a single fixed point in time
+# so that other date and time formats can be derived from it in an consistent way
+# (e.g. START_DATE_TIME_STRING matches START_DATE_TIME_NUMBER even if the second advances in between):
+readonly START_SECONDS=$( date +%s )
+readonly START_DATE_TIME_NUMBER=$( date -d@$START_SECONDS +%Y%m%d%H%M%S )
```

Comment:
How many seconds do you expect to see as having passed between this line and the preceding line where you set `START_SECONDS`?

I think that this is a bit of overkill 😄

Reply:
No it is not at all overkill.
Essentially it is the difference between right and wrong
(regardless that in this case doing it wrong would not cause harm
but see the 'cf.' below that in other cases there could be serious harm).

If one did simply
```
readonly START_DATE_TIME_NUMBER=$( date +%Y%m%d%H%M%S )
readonly START_DATE_TIME_STRING=$( date +'%F %T' )
```
the clock value could advance by one second between the two commands
so that $START_DATE_TIME_STRING could show
a bit later date than $START_DATE_TIME_NUMBER, cf.
https://github.com/rear/rear/blob/master/usr/share/rear/prep/NETFS/default/070_set_backup_archive.sh#L98
which explains that such "a bit later date" could be even the next day.

The probability that the clock value advances between the two commands is low
but that only means the probability is low that one could reproduce it
when a user reports the issue or when the user himself tries to make sense
what the reason behind could be when he was once hit by it.

Reply:
When turn of the year happens between two `date` commands
such "a bit later date" could be even the next year ;-)

Reply:
And about 20 and a half years ago
such "a bit later date" could have been the next millenium ;-))

Reply:
Better explanatory comment why one must not call the 'date' command several times
when one same point in time should be used for different date and time formats:
https://github.com/rear/rear/pull/2452/commits/01c21926e9ac695de917ec004870a41dc558b74e